lib/ts/recipe/emailpassword/utils.ts

```diff
@@ -338,12 +338,16 @@ export function mergeFormFields(
 }
 
 export function getFormattedFormField(field: NormalisedFormField): NormalisedFormField {
+    // Fields with the 'nonOptionalErrorMsg' property must have a valid message defined
+    if (field.optional === false && field.nonOptionalErrorMsg === "") {
```

Comment:
what if `optional` is `undefined`? Does that count as being `false` or `true`? Or will `flag.optional` never be `undefined` here?

Reply:
`flag.optional `will be `undefined` only if the user explicitly sets it
by default it will be `false`

it doesn't count as `false`

Reply:
I think undefined means that it is false.. so you have to check `if ((field.optional === false || field.optional === undefined) && field.nonOptionalErrorMsg === "")) {` right?